matiashoyld/icebreakers
# feat(simulation): Integrate LLM-Based Simulation and Expand Conversation Capabilities

## Changes
This PR replaces the hardcoded simulation steps with a dynamic LLM-powered conversation system and adds infrastructure for OpenAI integration. Key changes include:

### Infrastructure
- Added OpenAI client setup and basic request handlers
- Implemented utility functions for prompt generation and response parsing
- Added .gitignore rules for Cursor IDE configuration

### Simulation Logic
- Replaced hardcoded simulation steps with dynamic LLM-based generation
- Added conversation context support through text input field
- Implemented turn-taking logic with prompt templates
- Added dialogue history tracking for context awareness

### UI/UX Improvements
- Added text field for setting meeting topics
- Updated simulation controls to handle async LLM calls
- Added loading states during LLM processing
- Improved conversation flow visualization

### Code Organization
- Separated LLM-related utilities into dedicated modules
- Added type definitions for LLM responses and chat messages
- Removed hardcoded conversation data
- Added JSON parsing utilities for LLM responses

## Testing Instructions
1. Set up OpenAI API key in environment variables
2. Enter a conversation topic in the new text field
3. Start the simulation and observe the dynamic conversation flow
4. Verify that participants respond contextually to the chosen topic
